src/Raven.Server/Documents/Indexes/MapReduce/OutputToCollection/OutputReduceIndexWriteOperation.cs

```diff
@@ -60,7 +56,10 @@ public override void Commit(IndexingStatsScope stats)
         private async Task CommitOutputReduceToCollection()
         {
             foreach (var command in _outputReduceToCollectionCommandBatcher.CreateCommands())
+            {
+                DocumentDatabase.DatabaseShutdown.ThrowIfCancellationRequested();
```

Comment:
Do we need to call it explicitly here? Won't we get `TaskCancelledException` from `DocumentDatabase.TxMerger.Enqueue(command);` anyway?

Reply:
But we are disposing the "TxMerger" last so why we will get `TaskCancelledException`?

Reply:
Yeah, we're disposing it last but the database shutdown event is raised first:

https://github.com/ravendb/ravendb/blob/07f16e35386b93ddef7149b746e57734aec753b6/src/Raven.Server/Documents/DocumentDatabase.cs#L610

so I expected that an already enqueued command will result in the operation cancelled exception anyway (I haven't checked the code).

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-16243